sdk/tables/Azure.Data.Tables/src/TableEntityDictionary.cs

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Azure.Core;
+
+namespace Azure.Data.Tables
+{
+
+    public class TableEntityDictionary : IDictionary<string, object>
+    {
+        private readonly IDictionary<string, object> _properties;
+
+        /// <summary>
+        /// The partition key is a unique identifier for the partition within a given table and forms the first part of an entity's primary key.
+        /// </summary>
+        /// <value>A string containing the partition key for the entity.</value>
+        public string PartitionKey
+        {
+            get { return (string)_properties[TableConstants.PropertyNames.PartitionKey]; }
+            set { _properties[TableConstants.PropertyNames.PartitionKey] = value; }
+        }
+
+        /// <summary>
+        /// The row key is a unique identifier for an entity within a given partition. Together the <see cref="PartitionKey" /> and RowKey uniquely identify every entity within a table.
+        /// </summary>
+        /// <value>A string containing the row key for the entity.</value>
+        public string RowKey
+        {
+            get { return (string)_properties[TableConstants.PropertyNames.RowKey]; }
+            set { _properties[TableConstants.PropertyNames.RowKey] = value; }
+        }
+
+        /// <summary>
+        /// The Timestamp property is a DateTime value that is maintained on the server side to record the time an entity was last modified.
+        /// The Table service uses the Timestamp property internally to provide optimistic concurrency. The value of Timestamp is a monotonically increasing value,
+        /// meaning that each time the entity is modified, the value of Timestamp increases for that entity. This property should not be set on insert or update operations (the value will be ignored).
+        /// </summary>
+        /// <value>A <see cref="DateTimeOffset"/> containing the timestamp of the entity.</value>
+        public DateTimeOffset Timestamp
+        {
+            get { return (DateTimeOffset)_properties[TableConstants.PropertyNames.TimeStamp]; }
+            set { _properties[TableConstants.PropertyNames.TimeStamp] = value; }
```

Comment:
Oh woops, yeah that makes sense!!

Reply:
I looked back, and Track 1 does allow you to set it. I'm not really sure when it would ever come into play, but in my new commits, I took it out for now.